src/components/update-confirmation/update-confirmation-controller.ts

```diff
@@ -127,22 +126,27 @@ export async function removeMfaMethodConfirmationGet(
   req: Request,
   res: Response
 ): Promise<void> {
-  let message = req.t("pages.removeBackupMethod.confirm.message_unknown");
-  const removedMethod = (req.session.removedMfaMethods || []).find((m) => {
-    return "" + m.mfaIdentifier == req.query.id;
-  });
+  let message: string;
+  const removedMethod = req.session.removedMfaMethod;
+
+  if (!removedMethod) {
+    res.redirect(PATH_DATA.SECURITY.url);
+    return;
+  }
 
   if (removedMethod?.method.mfaMethodType === "AUTH_APP") {
```

Comment:
```suggestion
  if (removedMethod.method.mfaMethodType === "AUTH_APP") {
```
nit - we know `removedMethod` exists at this point. Same on line 141